tools/management/commands/get_gtp_ligands.py

```diff
@@ -0,0 +1,493 @@
+                    "ligand fetching error | module: fetch_publication. Row # is :")
+
+
+    def get_gpcrs(self):
+        target_list = list()
+        url = "https://www.guidetopharmacology.org/services/targets/families"
+        response = ''
+        while response == '':
+            try:
+                response = requests.get(url)
+
+            except:
+                print("Connection refused by the server..")
+                print("Let me sleep for 1 second")
+                print("ZZzzzz...")
+                time.sleep(1)
+                print("Was a nice sleep, now let me continue...")
+                response == ''
+
+        for entry in response.json():
+            try:
+                if entry['parentFamilyIds'] != None:
+                    if entry['parentFamilyIds'][0] == 694 or entry['parentFamilyIds'][0] == 115:
+                        target_list.extend(entry['targetIds'])
+            except:
+                print("Was a nice sleep, now let me continue...")
+        return target_list
+
+    def get_ligand_assays(self, targets):
+        assay_list = list()
+        response = requests.get(
+            "https://www.guidetopharmacology.org/services/interactions")
+        for entry in response.json():
+            try:
+                if entry['targetId'] != None and entry['targetId'] in targets:
+                    assay_list.append(entry)
+            except:
+                pass
+        return assay_list
+
+    def endogenous_ligands_from_db(self):
+        return GTP_endogenous_ligand.objects.all()
+
+    def convert_query_to_dict(self, endogenous_queryset):
+        endogenous_list = list()
+        for assay in endogenous_queryset:
+            single_dict = dict()
+            # import pdb; pdb.set_trace()
+            single_dict['id'] = assay
+            single_dict['ligand'] = assay.ligand
+            single_dict['ligand_type'] = assay.ligand_type
+            single_dict['endogenous_princip'] = assay.endogenous_princip
+            single_dict['receptor'] = assay.receptor
+            single_dict['pec50_avg'] = assay.pec50_avg
+            single_dict['pec50_min'] = assay.pec50_min
+            single_dict['pec50_max'] = assay.pec50_max
+            single_dict['pKi_avg'] = assay.pKi_avg
+            single_dict['pKi_min'] = assay.pKi_min
+            single_dict['pKi_max'] = assay.pKi_max
+            single_dict['gpt_link'] = assay.gpt_link
+            single_dict['references'] = list()
+            for publication in assay.publication.all():
+                single_dict['references'].append(publication)
+            endogenous_list.append(single_dict)
+        return endogenous_list
+
+    def process_assays(self, endogenous_list):
+        context = dict()
+        for assay in endogenous_list:
+            name = str(assay['ligand'].id) + \
+                '/' + str(assay['receptor'].id)
+            if name in context:
+                # import pdb; pdb.set_trace()
```

Comment:
If you set the if as a negative (if name not in context) you can initialize all the context lists inside the if and remove the duplicate part in this if/else by putting it outside the if

Reply:
- [done ]